crates/languages/src/typescript.rs

```diff
@@ -273,6 +273,17 @@ impl LspAdapter for EsLintLspAdapter {
             }
         }
 
+        let mut problems = json!({});
+
+        if let Some(problems_settings) = eslint_user_settings
+            .get("problems")
+            .and_then(|settings| settings.as_object())
+        {
+            if let Some(shorten_to_single_line) = problems_settings.get("shortenToSingleLine") {
+                problems["shortenToSingleLine"] = shorten_to_single_line.clone();
+            }
+        }
```

Comment:
Do we need to check for specific rules, or can we forward the complete `problems` object if it exists, otherwise an empty object?

Meaning: is the empty object (`{}`) a problem for ESLint?
```suggestion
        let problems = eslint_user_settings
            .get("problems")
            .cloned()
            .unwrap_or_else(|| json!({}));
```

Reply:
> but then I saw the codeActionOnSave code was unpacking things manaully (so I just followed that pattern).

Yeah, I think that one is different because by default we want to have `enabled: true`. 

> shouldn't be, the existing code is just passing an empty object to `problems`

Great!